### PR TITLE
show links to all luakit:// pages in the help page

### PR DESCRIPTION
**Pull request by jerojasro**
Hello,

I've re-installed luakit, and spent quite a good time trying to figure out how to configure it. I changed the documentation so it shows all the available chrome pages, so others don't have to dig that much to get to the config. facilities/new features available.

It's been a while since I wrote Lua; sorry for the lack of style.

Now the help page looks like this: https://imgur.com/rNt0fuV

### CODE REVIEW

File: lib/help_chrome.lua [hunk 1]
```diff
@@ -78,8 +85,24 @@ local index_html_template = [==[
 </body>
 ]==]
 
+local gen_html_chrome_pages = function()
+    local links = ""
+    for _, v in pairs(chrome.available_handlers()) do
```
Comment (aidanholm) [inline lib/help_chrome.lua]: should use `ipairs` instead of `pairs`

File: lib/chrome.lua [hunk 2]
```diff
@@ -170,6 +171,10 @@ local handlers = {}
 local on_first_visual_handlers = {}
 local page_funcs = {}
 
+function _M.available_handlers()
```
Comment (aidanholm) [inline lib/chrome.lua]: This function needs a documentation comment

**Review by aidanholm — CHANGES_REQUESTED** [review]
Thanks, this looks great! Just some minor changes needed

**Comment by jerojasro** [comment]
Thanks for checking. I've implemented the changes you suggested.

File: lib/chrome.lua [hunk 3]
```diff
@@ -171,6 +171,7 @@ local handlers = {}
 local on_first_visual_handlers = {}
 local page_funcs = {}
 
+--- Returns the currently registered luakit:// handlers, as an array of strings.
```
Comment (aidanholm) [inline lib/chrome.lua]: - Minor quibble about phrasing, and also needs a `@treturn` line:
```
--- Retrieve a list of the currently registered luakit:// handlers.
-- @treturn {string} A list of `luakit://` handler names, in alpabetical order.
```

**Review by aidanholm — CHANGES_REQUESTED** [review]
- Minor nit about documentation
- Please squash your fixup commits into the appropriate feature commit: the `ipairs` fixup would go into "show list of all available luakit:// pages in help page", and the doc fixup would go into the "chrome functionality one", producing two commits

Thanks!

**Comment by jerojasro** [comment]
I've added the suggested changes and force-pushed them; I'll try to keep in mind that rebasing/re-writing branches for PRs is OK in this project :)

**Comment by aidanholm** [comment]
Looks great, thanks!

Generally when you submit a PR (to any project), you do your work in a feature branch, and rebasing is fine because your branch is still separate from the main history, so nothing depends on it yet. In fact, rebasing is usually preferred, otherwise you end up with one or two meaningful commits and several fixups on top that clutter history. It's only once your work is merged that rebasing becomes a bad idea.

**Comment by McSinyx** [comment]
I'm sorry if it's not supposed to be asked here, but why is `luakit://` chosen over `about:` in the first place?

**Comment by McSinyx** [comment]
I'm sorry if it's not supposed to be asked here, but why was `luakit://` chosen over `about:` in the first place?

**Comment by aidanholm** [comment]
Not a problem. That was before my time, but it may have been because webkit already uses the `about:` scheme.

**Pull request merged**